cem/test/support-typedef-jsdoc-test.js

```diff
@@ -0,0 +1,138 @@
+import fs, { readFileSync } from 'fs';
+import { expect } from 'chai';
+import ts from 'typescript';
+import {
+  convertInterface,
+  findCustomType, findSubtypes, findTypePath, getConstructorNode, getTypesFromConstructor,
+} from '../support-typedef-jsdoc-utils.js';
+
+const filename = 'cem/test/fixtures/cc-test-component.js';
+const sourceCode = fs.readFileSync(filename, { encoding: 'utf-8' });
+// console.log(sourceCode);
+const sourceAst = ts.createSourceFile(filename, sourceCode, ts.ScriptTarget.ES2015, true);
+const classNode = sourceAst.statements.find((node) => node.kind === ts.SyntaxKind.ClassDeclaration);
+
+describe('getConstructorNode()', function () {
+  it('should get the constructor node.', function () {
+    const constructorNode = getConstructorNode(classNode, ts);
+    expect(constructorNode.kind).to.equal(ts.SyntaxKind.Constructor);
+  });
+});
+
+describe('getTypesFromConstructor()', function () {
+  const constructorNode = getConstructorNode(classNode, ts);
+
+  it('should retrieve the types present in the constructor.', function () {
+    const types = getTypesFromConstructor(constructorNode, ts);
+    expect(types).to.have.members(['Foo', 'Bar', 'TheInterface', 'TheType', 'TupleFoo', 'TupleBar']);
+  });
```

Comment:
Could we add another test with another file that only use one or two types from the file.